content/docs/rfcs/59/README.md

```diff
@@ -0,0 +1,185 @@
+    // Display name of the user
+    string display_name = 1;
+    // Description of the user
+    string description = 2;
+    // Color of the user title
+    string color = 3;
+}
+
+message URLData {
+    // Community, Channel, or User
+    bytes content = 1;
+}
+
+// Field on CommunityDescription, CommunityChat and ContactCodeAdvertisement (described in 56/STATUS-COMMUNITIES)
+message URLParams {
+    // Encoded URL data
+    string encoded_url_data = 1;
+    // Signature of encoded URL data, to ensure authenticity
+    string encoded_signature = 2;
+}
+```
+
+# Implementation
+
+The above wire format describes the data encoded in the URL. 
+The data MUST be serialized, compressed, and encoded using the following standards:
+
+## Encoding
+
+- [Base64url](https://datatracker.ietf.org/doc/html/rfc4648)
+
+## Compression
+
+- [Brotli](https://datatracker.ietf.org/doc/html/rfc7932)
+
+## Serialization
+
+- [Protocol buffers version 3](https://protobuf.dev/reference/protobuf/proto3-spec/)
+
+## Implementation Pseudocode
+
+### Encoding
+
+Encoding the URL MUST be done in the following order:
+
+```
+raw_data = {User | Channel | Community}
+serialized_data = protobuf_serialize(raw_data)
+compressed_data = brotli_compress(serialized_data)
+encoded_url_data = base64url_encode(compressed_data)
+```
+
+The `encoded_url_data` is then used to generate a signature using the private key.
+
+### Signature Generation
+
+Generation of the signature MUST be done by using the private key and url data:
+
+```
+signature = secp256k1_sign(encoded_url_data)
+encoded_signature = base64url_encode(signature)
+```
+
+After the signature is generated, the `encoded_url_data` and `encoded_signature` MUST be sent over the wire.
+
+### Decoding
+
+Decoding the URL MUST be done in the following order:
+
+```
+url_data = base64url_decode(encoded_url_data)
+signature = base64url_decode(encoded_signature)
+decompressed_data = brotli_decompress(url_data)
+deserialized_data = protobuf_deserialize(decompressed_data)
+raw_data = deserialized_data.content
+```
+
+The `raw_data` is then used to construct the appropriate data structure (User, Channel, or Community).
+
+### Signature Verification
+
+Verification of the public key MUST be done by using the deserialized signature and url data:
+
+```
+signature = base64url_decode(encoded_signature)
+url_data = base64url_decode(encoded_url_data)
+public_key = secp256k1_recover(signature, url_data)
+hash = keccak256(url_data)
+verified = secp256k1_verify(signature, hash, public_key)
+```
+
+The `verified` boolean denotes whether the signature is valid or not.
+
+## Example
+
+- See <https://github.com/status-im/status-web/pull/345/files>
+
+<!-- # (Further Optional Sections) -->
+
+# Discussions
+
+- See <https://github.com/status-im/status-web/issues/327>
+
+# Proof of concept
+
+- See <https://github.com/felicio/status-web/blob/825262c4f07a68501478116c7382862607a5544e/packages/status-js/src/utils/encode-url-data.compare.test.ts#L4>
```

Comment:
@prichodko about simplifying or solving the problem differently, I'm pointing this out here for better overview and future reference.

It's not the prettiest thing to navigate, but basically the results vary in

```js
{
      serialization: 'csv',
      compression: 'noop',
      encoding: 'encodeURIComponent',
    }
```

and

```js
expect(characterLength).toBe(184)
expect(encodedData).toHaveLength(243)
```

Previsouly, I shared with you https://github.com/status-im/status-web/pull/345#discussion_r1113129396. And with that, I'm still inclined towards

```js
{
        serialization: 'protobuf',
        compression: 'brotli',
        encoding: 'base64url',
      }
```